packages/block-editor/src/components/block-list/block.js

```diff
@@ -595,7 +595,11 @@ function BlockListBlock( {
 				</IgnoreNestedEvents>
 			</div>
 			{ showInserterShortcuts && (
-				<div className="editor-block-list__side-inserter block-editor-block-list__side-inserter">
+				<div
+					className="editor-block-list__side-inserter block-editor-block-list__side-inserter"
+					onFocus={ ( event ) => ( event.stopPropagation() ) }
+					tabIndex={ -1 }
```

Comment:
This tabIndex thing concerns me, is it needed?

Reply:
maybe I should have copied this comment everywhere:

`// While ideally it would be enough to capture the
					// bubbling focus event from the Inserter, due to the
					// characteristics of click focusing of `button`s in
					// Firefox and Safari, it is not reliable.
					//
					// See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus
					tabIndex={ -1 }`

Reply:
the problem is that if we don't mention it firefox still steals focus, from my testing at least

Reply:
Adding the comment seems like a good idea 👍